Applications/Tizen_native/CustomShortcut/src/data.c

```diff
@@ -297,18 +322,108 @@ void data_train_model() {
     goto CLEAN_UP;
   }
 
-  freopen("out.txt", "a+", stdout);
-
   status = ml_train_model_run(model, NULL);
   if (status != ML_ERROR_NONE) {
     LOG_E("run model failed %d", status);
     goto CLEAN_UP;
   }
 
 CLEAN_UP:
+  // restore stdout
+  fflush(stdout);
+  dup2(fd_stdout, 1);
+  close(fd_pipe);
```

Comment:
ditto

Reply:
Akaik, dup2 closes `newfd` to avoid race condition.